src/controllers/runtime/RuntimeMetadataController.ts

```diff
@@ -40,9 +40,17 @@ export default class RuntimeMetadataController extends AbstractController<Runtim
 	): Promise<void> => {
 		const hash = await this.getHashFromAt(at);
 
+		let historicApi;
+		if (at) {
+			historicApi = await this.api.at(hash);
+		}
+
+		const registry = historicApi ? historicApi.registry : this.api.registry;
+
 		RuntimeMetadataController.sanitizedSend(
 			res,
-			await this.service.fetchMetadata(hash)
+			await this.service.fetchMetadata(hash),
+			{ isMetadata: true, registry: registry }
```

Comment:
Just trying to understand the significance of this; so now, we are passing in a potentially different type registry from the one that will come along with the fetched metadata being provided?

Reply:
So for this, we actually are passing in the correct registry depending on whether the `at` query param is passed in. If there is no `at` we use the currently generated registry via `this.api`, and if `at` is used, then we retrieve the historic registry used at that block.